### DISCUSSION TITLE
Mac M1 npm ci build fail after 8ddc1fc7d (Electron related)

### DISCUSSION THREAD

**Discussion by kenykhung** [mozilla/hubs#5603] [Hubs Cloud Questions] [answered]
[2022-07-21T11_41_23_884Z-debug.log](https://github.com/mozilla/hubs/files/9159014/2022-07-21T11_41_23_884Z-debug.log)

Hi! I'm Ken,

I cannot run through `npm ci` since commit [8ddc1fc7d](https://github.com/mozilla/hubs/pull/5488)
Log is attached, looks like some modules (which i cannot identify from logs) are depended on a very old Electron version and it does not have arm64 support.

Any help would be greatly appreciated!
Thanks!
 
environment:
Mac M1 Pro 12.1
NPM 8.1.2
Node v16.15.0

**Comment by camelman90** [comment]
I have the same issue on Macbook Pro 14" M1, Monterey 12.4. You are right in the claim that the very old versions of electron is not built for arm64

Package dependency debt creeping up a little nowadays it seems. I´m gonna try to do my own independent research in the upcoming days to clean up the node packages and then push upstream if I have any luck.

**Reply by camelman90** [reply]
It looks like its both package files (hubs and admin panel) pulling dependencies:

@babel/preset-env -> browserslist -> electron-to-chromium

and

stylelint -> autoprefixer -> browserslist -> electron-to-chromium

I will be looking into upgrading babel preset-env and report back on results.

**Reply by camelman90** [reply]
From discord:

I thought it was some dependencies on electron-to-chromium, but after going through the tree with updated everything, (all the deep dependencies like browserslist, etc), it still didnt work. I did the elimination method by halving the package.json and running npm install. 

Removing the lib-hubs in package.json will let me run npm install without electron problems, on Node 16

**Reply by kenykhung** [reply]
@camelgod would you please guide me where to find the related discussion in discord? Thanks!

**Comment by matthewbcool** [comment]
Was able to run `npm ci` after downgrading to node 14.2

**Reply by camelman90** [reply]
Dont think this is a proper root cause solution, I am running fine on Node 16 without lib-hubs as described above. 
Something, somewhere is pulling in Electron 2.0.2.

**Reply by matthewbcool** [reply]
I agree

**Comment by kenykhung** [comment, marked as answer]
Based on @camelgod 's finding, i've submitted a PR #5621 which replace lib-hubs with three-particle-emitter.

**Reply by matthewbcool** [reply]
Thank you for that!

**Reply by keianhzo** [reply]
This PR should be fixing this issue: https://github.com/MozillaReality/lib-hubs/pull/46
